docs/architecture.md

```diff
@@ -0,0 +1,39 @@
+---
+title: Architecture
+---
+
+Architecture
+-------------
+
+### Components
+
+DirectCSI is made up of 4 components:
+
+| Component         | Description                                          |
+|-------------------|------------------------------------------------------|
+| CSI Driver        | Performs mounting, unmounting of provisioned volumes |
+| CSI Controller    | Schedules volumes on nodes                           |
+| Drive Controller  | Formats and manages drive lifecycle                  |
+| Volume Controller | manages volume lifecycle                             |
+
+The 4 components run as two different pods. 
+
+| Name                          | Components                                           | Description                        |
+|-------------------------------|------------------------------------------------------|------------------------------------|
+| DirectCSI Node Driver         | CSI Driver, Driver Controller, Volume Controller     | runs on every node as a DaemonSet  |
+| DirectCSI Central Controller  | CSI Controller                                       | runs as a deployment               |
+
+
+### Scalability
```

Comment:
Can we say that the default replica count for `DirectCSI Central Controller` is 3?